packages/core/lib/commands/create/helpers.js

```diff
@@ -70,81 +55,55 @@ function getLicense(options) {
   } catch {}
 }
 
-var Create = {
-  contract: function (directory, name, options, callback) {
-    if (typeof options === "function") {
-      callback = options;
-    }
-
-    var from = templates.contract.filename;
-    var to = path.join(directory, name + ".sol");
+const Create = {
+  contract: async function (directory, name, options) {
+    const from = templates.contract.filename;
+    const to = path.join(directory, name + ".sol");
 
     if (!options.force && fs.existsSync(to)) {
-      return callback(
-        new Error("Can not create " + name + ".sol: file exists")
-      );
+      throw new Error("Can not create " + name + ".sol: file exists");
     }
 
-    copy.file(from, to, function (err) {
-      if (err) return callback(err);
+    await promisify(copy.file)(from, to);
```

Comment:
Does this need `copy.file.bind(copy)`?

Reply:
promisification is expeeennsssivvve. We should never promisify the same function multiple times. I don't know what copy does here, but this should _probably_ be moved to the top of the file:

```javascript
const copy = require("../../copy");
const copyFile = promisify(copy.file.bind(copy));
```

Reply:
On second though: I just realized this promisification probably doesn't help truffle since these methods are likely only called a couple of times per Truffle invocation.